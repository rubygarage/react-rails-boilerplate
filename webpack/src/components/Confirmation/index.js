import React from 'react';
import { FormattedMessage } from 'react-intl';
import { Link } from 'react-router-dom';

function Confirmation() {
  return (
    <main>
      <div className="jumbotron">
        <h4> <FormattedMessage id="email_confirmation.success" /> </h4>

        <Link to="/sign_in">
          <p>
            <FormattedMessage id="email_confirmation.login" />
          </p>
        </Link>
      </div>
    </main>
  );
}

export default Confirmation;
