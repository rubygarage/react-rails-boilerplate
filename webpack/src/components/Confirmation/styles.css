.content {
  display: flex;
  flex-grow: 1;
  align-items: center;
  justify-content: center;
  padding-bottom: 13.5rem;
  max-width: 60rem;
  margin-left: auto;
  margin-right: auto;
}
