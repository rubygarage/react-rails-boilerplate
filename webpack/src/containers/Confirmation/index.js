import React from 'react';
import { connect } from 'react-redux';
import { emailConfirmation as emailConfirmationSaga } from 'sagas/emailConfirmation';
import ConfirmationComponent from 'components/Confirmation';

export class Confirmation extends React.PureComponent {
  render() {
    return (
      <ConfirmationComponent {...this.props} />
    );
  }
}

function preload(params, req, res) {
  const sagasToComplete = [];
  const confirmationToken = params.confirmation_token;

  sagasToComplete.push([emailConfirmationSaga, { confirmationToken }, res]) // eslint-disable-line

  return sagasToComplete;
}

Confirmation.preload = preload;

export default connect()(Confirmation);
