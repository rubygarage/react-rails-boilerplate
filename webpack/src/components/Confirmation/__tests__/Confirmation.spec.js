import React from 'react'
import { shallow } from 'enzyme'
import Confirmation from 'components/Confirmation'

describe('<Confirmation />', () => {
  const defaultProps = {
    loading: false
  }

  it('renders success message', () => {
    const component = shallow(
      <Confirmation {...defaultProps} />
    )

    expect(component).toMatchSnapshot()
  })
})
