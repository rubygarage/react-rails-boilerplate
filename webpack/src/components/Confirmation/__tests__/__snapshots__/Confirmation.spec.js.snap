// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Confirmation /> renders success message 1`] = `
<main>
  <div
    className="jumbotron"
  >
    <h4>
       
      <FormattedMessage
        id="email_confirmation.success"
        values={Object {}}
      />
       
    </h4>
    <Link
      replace={false}
      to="/sign_in"
    >
      <p>
        <FormattedMessage
          id="email_confirmation.login"
          values={Object {}}
        />
      </p>
    </Link>
  </div>
</main>
`;
